### PR TITLE
Move OBF provider from Joanie

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

The Open Badge Factory provider was originally developed for the https://github.com/openfun/joanie project. Since we also want to use this provider in other project, we decided to distribute open badge provider clients as a separated package.

## Proposal

- [x] move providers from Joanie to the `obc`
- [x] remove Django dependency for testing
- [x] add CircleCi configuration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,79 @@
...
+target to build the Docker development image:
+
+```
+$ make bootstrap
+```
+
+To run tests, type:
+
+```
+$ make tests
```
Comment (quitterie-lcs) [inline README.md:57]:
```suggestion
$ make test
```

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,87 @@
+;;
+;; Ralph package
```
Comment (quitterie-lcs) [inline setup.cfg:2]:
```suggestion
;; Open badge client package
```

File: src/obc/providers/obf.py [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+        if self.raise_for_status:
+            response.raise_for_status()
+
+        return response
+
+
+class Badge(BaseModel):
+    """Open Badge Factory Badge Model."""
+
+    id: NoneStr = None
```
Comment (quitterie-lcs) [inline src/obc/providers/obf.py:183]: Is this type enough for the id ?

File: src/obc/providers/obf.py [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+        if is_created and id_ is None:
+            raise ValidationError("Created badges should have an `id` field.")
+
+        return values
+
+
+class BadgeQuery(BaseModel):
+    """Open Badge Factory badge query filters."""
+
+    draft: Optional[Literal[0, 1]] = None
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/obc/providers/obf.py:215]: This is specified in Open Badge Factory API. Should be a boolean I agree, but...

File: src/obc/providers/obf.py (hunk 3 above)
Comment (jmaupetit) [reply src/obc/providers/obf.py:183]: Yes, OBF badge identifier are not "standards", _e.g._ `RPG8PZa4J4aDD`. Not sure we want to be pedantic on this.

File: src/obc/providers/obf.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/obc/providers/obf.py:215]: Ah ok :smile:

**Review by quitterie-lcs — APPROVED** [review]

**Pull request merged**